src/Razor/src/Microsoft.NET.Sdk.Razor/ValidateStaticWebAssetsUniquePaths.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class ValidateStaticWebAssetsUniquePaths : Task
+    {
+        private const string BasePath = "BasePath";
+        private const string RelativePath = "RelativePath";
+        private const string TargetPath = "TargetPath";
+
+        [Required]
+        public ITaskItem[] StaticWebAssets { get; set; }
+
+        [Required]
+        public ITaskItem[] WebRootFiles { get; set; }
+
+        public override bool Execute()
+        {
+            var assetsByWebRootPaths = new Dictionary<string, ITaskItem>(StringComparer.OrdinalIgnoreCase);
+            for (var i = 0; i < StaticWebAssets.Length; i++)
+            {
+                var contentRootDefinition = StaticWebAssets[i];
+                if (!EnsureRequiredMetadata(contentRootDefinition, BasePath) ||
+                    !EnsureRequiredMetadata(contentRootDefinition, RelativePath))
+                {
+                    return false;
+                }
+                else
+                {
+                    var webRootPath = GetWebRootPath(
+                        contentRootDefinition.GetMetadata(BasePath),
+                        contentRootDefinition.GetMetadata(RelativePath));
+
+                    if (assetsByWebRootPaths.TryGetValue(webRootPath, out var existingWebRootPath) &&
+                        !string.Equals(contentRootDefinition.ItemSpec, existingWebRootPath.ItemSpec, StringComparison.OrdinalIgnoreCase))
```

Comment:
Personal preference, but it's much easier to reason about this if this was formatted as so: